rustls-platform-verifier/src/verification/android.rs

```diff
@@ -305,4 +284,38 @@ impl ServerCertVerifier for Verifier {
             }
         }
     }
+
+    fn verify_tls12_signature(
+        &self,
+        message: &[u8],
+        cert: &pki_types::CertificateDer<'_>,
+        dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, TlsError> {
+        verify_tls12_signature(
+            message,
+            cert,
+            dss,
+            &rustls::crypto::ring::default_provider().signature_verification_algorithms,
```

Comment:
Would it be better to store the provider on the verifier structure so it can be referenced through `&self`? Otherwise this is going to re-allocate two vecs on each signature (`cipher_suites` and `kx_groups`).

This comment also applies to the other platform verifier implementations.

Reply:
Fixed.